Effects/AlphaDistort.fx

```diff
@@ -0,0 +1,39 @@
+﻿sampler uImage0 : register(s0);
+sampler uImage1 : register(s1);
+sampler uImage2 : register(s2);
+float3 uColor;
+float3 uSecondaryColor;
+float uOpacity;
+float uSaturation;
+float uRotation;
+float uTime;
+float4 uSourceRect;
+float2 uWorldPosition;
+float uDirection;
+float3 uLightSource;
+float2 uImageSize0;
+float2 uImageSize1;
+
+float power;
+float speed;
+float opacity;
+
+float2 offset;
+
+float4 drawColor;
+
+float4 PixelShaderFunction(float2 coords : TEXCOORD0) : COLOR0
+{
+	float2 off = float2(0, sin(uTime + (coords.x + offset.x) * speed) * sin(1.57 + (coords.x + offset.x) * 0.37 * speed - uTime) * sin((coords.x + offset.x) * 0.21 * speed - uTime) * power);
```

Comment:
There are inconsistent tabs/spaces here. Additionally, it may be wise to split some of the calculations here out to seperate variables to make it more readable